syntax = "proto3";

package renderer;

message ShaderConfig {
  string name = 1;
  // Render Shader
  optional string vs_path = 2;
  optional string fs_path = 3;
  optional string gs_path = 4;
  optional string ts_path = 5;
  // Compute Shader
  optional string cs_path = 6;
}

enum TextureType {
  Unknown = 0;
  Texture2D = 1;
  Cubemap = 2;
  Texture2DArray = 3;
}

message TextureConfig {
  string name = 1;
  int32 level_num = 2;
  TextureType texture_type = 3;
  repeated string path = 4;
}

message ModelConfig {
  string name = 1;
  string model_dir = 2;
  string model_file = 3;
}

message ConfigData {
  repeated ShaderConfig shader_config = 1;
  repeated TextureConfig texture_config = 2;
  repeated ModelConfig model_config = 3;
}